components/calendar/src/calendar_arithmetic.rs

```diff
@@ -382,22 +382,33 @@ impl<C: CalendarArithmetic> ArithmeticDate<C> {
     where
         C: CalendarArithmetic<YearInfo = ()>,
     {
-        let max_month = C::months_for_every_year(year, ());
+        Self::new_from_ordinals_with_info(year, month, day, ())
+    }
+
+    /// Construct a new arithmetic date from a year, month ordinal, and day, bounds checking
+    /// the month and day
+    pub fn new_from_ordinals_with_info(
+        year: i32,
+        month: u8,
+        day: u8,
+        info: C::YearInfo,
+    ) -> Result<Self, CalendarError> {
+        let max_month = C::months_for_every_year(year, info);
         if month > max_month {
             return Err(CalendarError::Overflow {
                 field: "month",
                 max: max_month as usize,
             });
         }
-        let max_day = C::month_days(year, month, ());
+        let max_day = C::month_days(year, month, info);
         if day > max_day {
             return Err(CalendarError::Overflow {
                 field: "day",
                 max: max_day as usize,
             });
         }
 
-        Ok(Self::new_unchecked(year, month, day))
+        Ok(Self::new_unchecked_with_info(year, month, day, info))
```

Comment:
Nit: Can you debug-assert that `info` is the correct info for `year`? That's an invariant of CalendarArithmetic that isn't being enforced in this constructor as far as I see.

Reply:
Will do. In the long term I want CalendarArithmetic's methods to always take `ArithmeticDate` (even though it means in some cases we'll need to make fake dates)

Reply:
This is only possible at call-site, CalendarArithmetic does not know how to construct a YearInfo

Reply:
Yeah, this isn't possible to do; CalendarArithmetic doesn't know anything about YearInfo.